### PR TITLE
Support prometheus.remoteWrite flag

### PR DESCRIPTION
**Pull request by cthtrifork**

### CODE REVIEW

File: charts/oaas-observability/templates/prometheus/prometheus.yaml [hunk 1]
```diff
@@ -198,7 +198,12 @@ spec:
 {{- end }}
 {{- if or .Values.prometheus.prometheusSpec.containers .Values.prometheus.agent }}
```
Comment (langecode) [inline charts/oaas-observability/templates/prometheus/prometheus.yaml:199]:
```suggestion
{{- if or .Values.prometheus.prometheusSpec.containers .Values.prometheus.agent .Values.prometheus.remoteWrite }}
```

File: charts/oaas-observability/Chart.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ description: A Helm chart to deploy obeservability stack on Kubernetes
 home: https://github.com/neticdk/k8s-oaas-observability
 sources:
   - https://github.com/neticdk/k8s-oaas-observability
-version: 2.2.0
+version: 2.3.0
```
Comment (langecode) [inline charts/oaas-observability/Chart.yaml:8]: I think this would be ok to just qualify as patch version.

```suggestion
version: 2.2.1
```

**Review by langecode — CHANGES_REQUESTED** [review]
I am not quite sure what the use-case would be for this but I do think it looks ok - with the minor adjustments proposed.

**Comment by cthtrifork** [comment]
> I am not quite sure what the use-case would be for this but I do think it looks ok - with the minor adjustments proposed.

The use case is to allow `remote_write`. We have some IoT agent which are not always running and we can not pull from. They should send/push the metrics to the `remote_write` endpoint. I have made an ingress that is bound to the specific url of the `remote_write` url of our prometheus instance + added basic auth to the ingress.

**Comment by cthtrifork** [comment]
I had to rebase. Let me bump patch again

**Comment by langecode** [comment]
> The use case is to allow `remote_write`. We have some IoT agent which are not always running and we can not pull from. They should send/push the metrics to the `remote_write` endpoint. I have made an ingress that is bound to the specific url of the `remote_write` url of our prometheus instance + added basic auth to the ingress.

I see no problem in making the chart here support setting up a remote write receiver. However, the "goto" solution for the use case you describe would normally be to set up a Prometheus [pushgateway](https://github.com/prometheus/pushgateway) - there is support for flushing data to push-gateway from multiple libraries, i.e., the .NET library for prometheus.

The remote write protocol by default allows you to push any sample directly into the tsdb of Prometheus - and you probably do not want a single IoT client to be able to mess with your data like that. What's even more in a setup at scale where Prometheus passes data to a central tsdb (like Cortex or VictoriaMetrics) the data ingested by remote write would by default be passed onto the central database.

The remote write protocol is designed for federation like if you want to pass Prometheus data into a central database. It was not designed to handle sampling of data from individual endpoints.

But we would be happy to accept the PR with the small changes suggested - the code changes looks quite ok.

**Review by langecode — APPROVED** [review]

**Pull request merged**